compiler/rustc_passes/src/check_attr.rs

```diff
@@ -1782,6 +1783,17 @@ impl<'tcx> CheckAttrVisitor<'tcx> {
         let mut is_simd = false;
         let mut is_transparent = false;
 
+        let has_align =
+            hints.iter().find_map(|i| (i.name_or_empty() == sym::align).then(|| i.span()));
+        let (align_note, align_present_note) =
```

Comment:
`align_present_note` and `align_note` are mutually exclusive, and both are only conditional on `self.tcx.features().fn_align`. I'd prefer if you represent this with an:

``` 
#[derive(Subdiagnostic)]
enum FnAlignSuggestion {
    #[diag(passes_align_present_note)]
    OnlyAlign,
    #[diag(passes_align_note)]
    SupportsAlign,
}
```

And then using `Option<FnAlignSuggestion>`. This avoid needing to add *two* fields to every diagnostic.

This is kinda unrelated to your change here, though -- I'd somewhat rather you break this into a separate PR so we can refine it separately.

Reply:
thanks now I also understand properly how to use subdiagnostics like this :)